VSRAD.Package/Server/CommunicationChannel.cs

```diff
@@ -81,11 +107,56 @@ public CommunicationChannel(SVsServiceProvider provider, IProject project)
             _project = project;
             _project.RunWhenLoaded((options) =>
                 options.PropertyChanged += (s, e) => { if (e.PropertyName == nameof(options.ActiveProfile)) ForceDisconnect(); });
+            var match = _extensionVersionRegex.Match(typeof(IProject).Assembly.CodeBase);
+            if (!match.Success)
+                throw new Exception("Error while getting current extension version.");
+            _extensionVersion = new Version(match.Groups["version"].Value);
         }
 
         public Task<T> SendWithReplyAsync<T>(ICommand command) where T : IResponse =>
             SendWithReplyAsync<T>(command, tryReconnect: true);
 
+        private async Task<bool> TryProcessClientHandshake(TcpClient client)
+        {
+            StreamWriter writer = new StreamWriter(client.GetStream(), Encoding.UTF8) { AutoFlush = true };
+            StreamReader reader = new StreamReader(client.GetStream(), Encoding.UTF8);
+
+            // Send client version to server
+            //
+            await writer.WriteLineAsync(_extensionVersion.ToString()).ConfigureAwait(false);
+            // Obtain server version
+            //
+            String serverResponse = await reader.ReadLineAsync().ConfigureAwait(false);
+            Version serverVersion = null;
+            if (!Version.TryParse(serverResponse, out serverVersion))
+            {
+                // Inform server that client declines serve's version
+                //
+                await writer.WriteLineAsync(HandShakeStatus.client_not_accepted.ToString()).ConfigureAwait(false);
```

Comment:
Here and below: case of `HandShakeStatus` changed only in definition, but not in usages